packages/web-components/src/drawer/drawer.ts

```diff
@@ -0,0 +1,244 @@
+
+  public disconnectedCallback(): void {
+    super.disconnectedCallback();
+  }
+
+  /**
+   * The dialog element.
+   * @public
+   *
+   */
+  @observable
+  public dialog!: HTMLDialogElement;
+
+  /**
+   * Determines whether the drawer should be displayed as modal, non-modal, or alert.
+   * When in modal or alert mode, an overlay is applied over the rest of the view.
+   * @public
+   * @remarks
+   * HTML Attribute: modalType
+   */
+  @attr({ attribute: 'modal-type' })
+  public modalType: DrawerModalType = DrawerModalType.modal;
+
+  /**
+   * The ID of the element that labels the drawer.
+   * @public
+   * @remarks
+   * HTML Attribute: aria-labelledby
+   */
+  @attr({ attribute: 'aria-labelledby' })
+  public ariaLabelledby?: string;
+
+  /**
+   * The ID of the element that describes the drawer.
+   * @public
+   * @remarks
+   * HTML Attribute: aria-describedby
+   */
+  @attr({ attribute: 'aria-describedby' })
+  public ariaDescribedby?: string;
+
+  /**
+   * Sets the type of the drawer (overlay/inline).
+   * @public
+   * @remarks
+   * HTML Attribute: type
+   * @defaultValue overlay
+   */
+  @attr
+  public type?: DrawerType;
+
+  /**
+   * Sets the position of the drawer (start/end).
+   * @public
+   * @remarks
+   * HTML Attribute: position
+   * @defaultValue start
+   */
+  @attr
+  public position?: DrawerPosition;
+
+  /**
+   * Sets the size of the drawer (small/medium/large).
+   * @public
+   * @remarks
+   * HTML Attribute: size
+   * @defaultValue medium
+   */
+  @attr({ attribute: 'size' })
+  public size?: DrawerSize;
+
+  /**
+   * Sets the open state of the drawer
+   * @public
+   * @remarks
+   * HTML Attribute: open
+   * @defaultValue false
+   */
+  @attr({ mode: 'boolean' })
+  public open: boolean = false;
+
+  /**
+   * Emits a 'cancel' event.
+   * @internal
+   */
+  public cancel(): void {
+    this.$emit('cancel');
+  }
+
+  /**
+   * Handles changes to the `open` property.
+   * @public
+   */
+  public openChanged(oldValue: boolean, newValue: boolean): void {
+    if (newValue !== oldValue) {
+      if (newValue && !oldValue) {
+        this.show();
+      } else if (!newValue && oldValue) {
+        this.hide();
+      }
+    }
+    this.$emit('onOpenChange', { open: this.open });
+  }
+
+  /**
+   * @public
+   * Method called when the 'modalType' attribute changes
+   */
+  public typeChanged(oldValue: DrawerType, newValue: DrawerType): void {
+    if (newValue !== oldValue) {
+      if (newValue == DrawerType.inline) {
+        this.modalType = DrawerModalType.nonModal;
+      }
+    }
+  }
+
+  /**
+   * Sets the component state based on the `open` property.
+   * @public
+   */
+  public setComponent(): void {
+    if (this.open) {
+      this.show();
+    }
+  }
+
+  /**
+   * Opens the drawer if it is currently hidden.
+   * @public
+   */
+  public show(): void {
+    if (!this.dialog.open) {
+      if (this.type === DrawerType.inline || this.modalType === DrawerModalType.nonModal) {
```

Comment:
not sure that using two separate types is not complicating things. it seems to allow confusing state combinations like inline+modal or overlay+onModal

Reply:
I do agree here. There is an `overlay + nonModal` state combination which is used for when consumer need the drawer but still need to interact with the rest of the page.

But an `inline + modal` drawer does not exist. Currently, if the user sets the drawer to inline then the drawer is automatically set to a `non-modal` type. Not sure how else to handle this as per the distinction in the design spec.